src/esolver/lis_esolver_ai.c

```diff
@@ -255,7 +255,7 @@ LIS_INT lis_eai(LIS_ESOLVER esolver)
       i = i + 1;
 
       /* eigenvalues on the diagonal of H */
-      if (fabs(h[i+(i-1)*ss])<tol)
```

Comment:
`h` is a square matrix of size `ss`.
In some cases when `i=ss`, `i+(i-1)*ss=ss*ss`, and later `i+i*ss=ss*ss+ss`, which both exceed the length of `h`.
This check does not apply to the last diagonal of the Hessenberg matrix.